app/config/wiring.scala

```diff
@@ -16,73 +16,24 @@
-
-  override protected def appNameConfiguration: Configuration = Play.current.configuration
-}
-
-trait WSHttp extends HttpGet with WSGet with HttpPut with WSPut with HttpPost with WSPost with HttpDelete with WSDelete with AppName
-
-object WSHttp extends WSHttp {
-  override val hooks = NoneRequired
-
-  override protected def actorSystem: ActorSystem = Play.current.actorSystem
-
-  override protected def configuration: Option[Config] = Some(Play.current.configuration.underlying)
-
-  override protected def appNameConfiguration: Configuration = Play.current.configuration
-}
-
-object WSHttpWithAudit extends WSHttp with HttpHooks with HttpAuditing {
-  override val hooks = Seq(AuditingHook)
-  override lazy val auditConnector: Auditing = AtedSubscriptionFrontendAuditConnector
-
-  override protected def actorSystem: ActorSystem = Play.current.actorSystem
-
-  override protected def configuration: Option[Config] = Some(Play.current.configuration.underlying)
-
-  override protected def appNameConfiguration: Configuration = Play.current.configuration
-}
-
-object CachedStaticHtmlPartialProvider extends CachedStaticHtmlPartialRetriever {
-  override val httpGet = WSHttp
+class AtedSubscriptionFrontendAuditConnector @Inject()(conf: ServicesConfig,
+                                                       auditConf: AuditingConfigProvider) extends AuditConnector {
+  override lazy val auditingConfig: AuditingConfig = auditConf.get()
 }
 
-object AtedSessionCache extends SessionCache with AppName with ServicesConfig {
-  override lazy val http = WSHttp
-  override lazy val defaultSource = appName
-  override lazy val baseUri = baseUrl("cachable.session-cache")
-  override lazy val domain = getConfString("cachable.session-cache.domain", throw new Exception(s"Could not find config 'cachable.session-cache.domain'"))
-
-  override protected def appNameConfiguration: Configuration = Play.current.configuration
-
-  override protected def mode: Mode = Play.current.mode
+class CachedStaticHtmlPartialProvider @Inject()(val httpGet: DefaultHttpClient) extends CachedStaticHtmlPartialRetriever
 
-  override protected def runModeConfiguration: Configuration = Play.current.configuration
-}
-
-object AuthClientConnector extends PlayAuthConnector with ServicesConfig {
-    val serviceUrl: String = baseUrl("auth")
-    lazy val http = WSHttp
-
-  override protected def mode: Mode = Play.current.mode
-
-  override protected def runModeConfiguration: Configuration = Play.current.configuration
-}
+class AtedSessionCache @Inject()(conf: ServicesConfig,
+                                 val http: DefaultHttpClient) extends SessionCache {
+  lazy val defaultSource: String = conf.getConfString("appName", "ated-subscription-frontend")
+  lazy val baseUri: String = conf.baseUrl("cachable.session-cache")
+  lazy val domain: String = conf.getConfString("cachable.session-cache.domain", throw new Exception(s"Could not find config 'cachable.session-cache.domain'"))
```

Comment:
Changing this to `getString` means we don't need to throw an exception in our code.
These look better placed in ApplicationConfig too.